### ISSUE TITLE
Option to Store Project Configuration in Project Folder (or Vault).

### ISSUE THREAD

**Issue by jeetsukumaran** [marcusolsson/obsidian-projects#416]
### What would you like to be added?

I would like to have the option to be able to store the project configuration outside of the `.obsidian` hierarchy.

One approach might be a plugin-wide or per-project setting, where the user can choose for "Project Configuration Data": "default" ("data.json" in plug-in directory), "project folder" (for projects with data source in the folder), or "in folder I specify below" (e.g, "__annex__/projects/data").

### Why is this needed?

I version-control my vault and vault configuration (``.obsidian/**``) separately. This allows me to sync the configuration across different vaults easily (I ``.gitignore`` the Obsidian workspace data ```.obsidian/workspace``). I can add this plugin's ``data.json`` to the ``./obsidian/.gitignore`` file as well. But it would be nice to version-control/backup the project configuration as part of the Vault content as needed.

I **LOVE** the [design philosophy](https://github.com/marcusolsson/obsidian-projects#design-philosophy) of this project and recognize that this FR might violate the "Leave No Trace" principle. But with my suggestion, the clutter remains outside of notes, and can be seen as a sort of middle-level "infrastructure" of a vault like templates etc.
 
With this feature, at least for selected projects (with particularly complex setups), we can version-control/backup the configuration cleanly, maintaining connection to the content (e.g., fields, data sources ).

**Comment by GoshaEgorian** [comment]
The same for me here

**Comment by marcusolsson** [comment]
I don't think this violates the design principle. Like you say, this doesn't affect the notes themselves. This seems to simply be a matter of where to store the project configuration file.

I agree we should solve this, but I'd like to make sure the solution aligns with #77.

**Comment by s-taylor** [comment]
I agree with this 100%. I use projects on multiple devices and having to reconfigure each project on each device is annoying.

**Comment by Doodidan** [comment]
> I agree with this 100%. I use projects on multiple devices and having to reconfigure each project on each device is annoying.

@s-taylor For now to sync project settings you need to sync filre `.obsidian/plugins/obsidian-projects/data.json`

If you use git, add this to `.gitignore` at root:
```
.obsidian/*
!.obsidian/plugins/
.obsidian/plugins/*
!.obsidian/plugins/obsidian-projects/
.obsidian/plugins/obsidian-projects/*
!.obsidian/plugins/obsidian-projects/data.json
```
